sky/clouds/service_catalog/common.py

```diff
@@ -146,19 +150,36 @@ def get_hourly_cost_impl(
     df: pd.DataFrame,
     instance_type: str,
     region: Optional[str],
+    zone: Optional[str],
     use_spot: bool = False,
 ) -> float:
-    """Returns the cost, or the cheapest cost among all zones for spot."""
-    df = _get_instance_type(df, instance_type, region)
-    assert pd.isnull(
-        df['Price'].iloc[0]) is False, (f'Missing price for "{instance_type}, '
-                                        f'Spot: {use_spot}" in the catalog.')
-    # TODO(zhwu): We should handle the price difference among different regions.
-    price_str = 'SpotPrice' if use_spot else 'Price'
-    assert region is None or len(set(df[price_str])) == 1, df
+    """Returns the hourly price of a VM instance in the given region and zone.
 
-    cheapest_idx = df[price_str].idxmin()
+    Refer to get_hourly_cost in service_catalog/__init__.py for the docstring.
+    """
+    df = _get_instance_type(df, instance_type, region, zone)
+    if df.empty:
+        if zone is None:
+            if region is None:
+                region_or_zone = 'all regions'
+            else:
+                region_or_zone = f'region {region!r}'
+        else:
+            region_or_zone = f'zone {zone!r}'
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(f'Instance type {instance_type!r} not found '
+                             f'in {region_or_zone}.')
+
+    # If the zone is specified, only one instance type should be returned.
```

Comment:
instance type -> price?

Reply:
Changed to `# If the zone is specified, only one row should be found by the query.`